File: Server/Models/Response/BaseResponse.py
from pydantic import BaseModel


class BaseResponse(BaseModel):
    success: bool
    message: str

    @staticmethod
    def success(message: str) -> BaseResponse:
        return BaseResponse(success=True, message=message)

    @staticmethod
    def failure(message: str) -> BaseResponse:
        return BaseResponse(success=False, message=message)

File: Server/src/global/models/base.response.ts
export default class BaseResponse {
    private constructor(
        public readonly success: boolean,
        public readonly message: string,
    ) {}

    public static success(message: string): BaseResponse {
        return new BaseResponse(true, message);
    }
    public static failure(message: string): BaseResponse {
        return new BaseResponse(false, message);
    }
}
